File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/DoublePropertySerializer.java
package ru.timeconqueror.timecore.common.capability.property.serializer;

import net.minecraft.nbt.CompoundTag;

public class DoublePropertySerializer implements IPropertySerializer<Double>{
    public static final IPropertySerializer<Double> INSTANCE = new DoublePropertySerializer();
    public static final IPropertySerializer<Double> NULLABLE = IPropertySerializer.nullable(INSTANCE);

    @Override
    public void serialize(String name, Double value, CompoundTag nbt) {
        nbt.putDouble(name, value);
    }

    @Override
    public Double deserialize(String name, CompoundTag nbt) {
        return nbt.getDouble(name);
    }
}

File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/DoublePropertySerializer.kt
package ru.timeconqueror.timecore.common.capability.property.serializer

import net.minecraft.nbt.CompoundTag


object DoublePropertySerializer : IPropertySerializer<Double> {

    override fun serialize(name: String, value: Double, nbt: CompoundTag) = nbt.putDouble(name, value)
    override fun deserialize(name: String, nbt: CompoundTag) = nbt.getDouble(name)

    object Nullable : NullPropertySerializer<Double>(this)
}